pycomlink/processing/nearby_rain_retrival.py

```diff
@@ -147,45 +146,46 @@ def nearby_rainfall_retrival(
 
             a, b = [], []
             for i_freq, i_pol in zip(f_GHz.flatten(), pol.flatten()):
-                a_tmp, b_tmp = a_b(f_GHz=i_freq, pol=i_pol,
-                                   approx_type=a_b_approximation)
+                a_tmp, b_tmp = a_b(
+                    f_GHz=i_freq, pol=i_pol, approx_type=a_b_approximation
+                )
                 a.append(a_tmp)
                 b.append(b_tmp)
             a = np.reshape(np.array(a), shape_save)
             b = np.reshape(np.array(b), shape_save)
 
             # turn a and b values to xarray.DataArray and check whether
             # no or the dim channel_id is available
-            if 'channel_id' in list(pref.dims):
-                a = xr.DataArray(a, coords=dict(cml_id=pref.cml_id,
-                                                channel_id=pref.channel_id))
-                b = xr.DataArray(b, coords=dict(cml_id=pref.cml_id,
-                                                channel_id=pref.channel_id))
+            if "channel_id" in list(pref.dims):
+                a = xr.DataArray(
+                    a, coords=dict(cml_id=pref.cml_id, channel_id=pref.channel_id)
+                )
+                b = xr.DataArray(
+                    b, coords=dict(cml_id=pref.cml_id, channel_id=pref.channel_id)
+                )
             else:
                 a = xr.DataArray(a, coords=dict(cml_id=pref.cml_id))
                 b = xr.DataArray(b, coords=dict(cml_id=pref.cml_id))
 
         else:
-            raise IndexError(
-                "Size of `f_GHz` and `pol` must be identical."
-            )
+            raise IndexError("Size of `f_GHz` and `pol` must be identical.")
 
-    elif (a is not None) and (b is not None) and (f_GHz is None) and (
-            pol is None):
+    elif (a is not None) and (b is not None) and (f_GHz is None) and (pol is None):
         # in this case we use `a` and `b` from args
         pass
     else:
         raise ValueError(
-            "Either `f_GHz` and `pol` or `a` and `b` have to be passed. Any other combination is not allowed."
+            "Either `f_GHz` and `pol` or `a` and `b` have to be passed. Any "
+            "other combination is not allowed."
         )
 
     # calculate minimum and maximum rain-induced attenuation
     A_min = pref - p_c_max
     A_max = pref - p_c_min
 
     # retrieve rainfall intensities
-    r_min = ((1 / a) ** (1 / b) * (((A_min - waa_max) / length))) ** b
-    r_max = ((1 / a) ** (1 / b) * (((A_max - waa_max) / length))) ** b
+    r_min = ((A_min - waa_max)/(a*length))**(1/b)
+    r_max = ((A_max - waa_max)/(a*length))**(1/b)
 
     # not allowing negative rain intensities
     r_min = xr.where(A_min - waa_max < 0, 0, r_min)
```

Comment:
Wouldn't it be clearer if you just write `r_min = xr.where(r_min < 0, 0, r_min)` here? Same for the next line.

Reply:
Yes indeed. Changed it.

Reply:
`r_min = xr.where(r_min < 0, 0, r_min)` is not the same as `rmin` and `rmax` are not negative when `A_min - waa_max` or `A_max - waa_max` are negative but `nan`. Therefore, now first wet antenna is removed, than negative `A_min` and `A_max` are set to 0 and then the `k-R` relation is used to calculate rain rates.